src/robusta/core/discovery/discovery.py

```diff
@@ -70,87 +72,132 @@ def discovery_process() -> DiscoveryResults:
+                            extract_containers(statefulset),
+                            extract_volumes(statefulset),
+                            extract_total_pods(statefulset),
+                            extract_ready_pods(statefulset),
+                        )
+                        for statefulset in statefulsets.items
+                    ]
+                )
+                continue_ref = statefulsets.metadata._continue
+                if not continue_ref:
+                    break
+
+            # discover daemonsets
+            continue_ref = None
+            for _ in range(DISCOVERY_MAX_BATCHES):
+                daemonsets: V1DaemonSetList = client.AppsV1Api().list_daemon_set_for_all_namespaces(
+                    limit=DISCOVERY_BATCH_SIZE, _continue=continue_ref
+                )
+                active_services.extend(
+                    [
+                        Discovery.__create_service_info(
+                            daemonset.metadata,
+                            "DaemonSet",
+                            extract_containers(daemonset),
+                            extract_volumes(daemonset),
+                            extract_total_pods(daemonset),
+                            extract_ready_pods(daemonset),
+                        )
+                        for daemonset in daemonsets.items
+                    ]
+                )
+                continue_ref = daemonsets.metadata._continue
+                if not continue_ref:
+                    break
+
+            # discover replicasets
+            continue_ref = None
+            for _ in range(DISCOVERY_MAX_BATCHES):
+                replicasets: V1ReplicaSetList = client.AppsV1Api().list_replica_set_for_all_namespaces(
+                    limit=DISCOVERY_BATCH_SIZE, _continue=continue_ref
+                )
+                active_services.extend(
+                    [
+                        Discovery.__create_service_info(
+                            replicaset.metadata,
+                            "ReplicaSet",
+                            extract_containers(replicaset),
+                            extract_volumes(replicaset),
+                            extract_total_pods(replicaset),
+                            extract_ready_pods(replicaset),
+                        )
+                        for replicaset in replicasets.items
+                        if not replicaset.metadata.owner_references and replicaset.spec.replicas > 0
+                    ]
+                )
+                continue_ref = replicasets.metadata._continue
+                if not continue_ref:
+                    break
+
+            # discover pods
+            continue_ref = None
+            for _ in range(DISCOVERY_MAX_BATCHES):
+                pods: V1PodList = client.CoreV1Api().list_pod_for_all_namespaces(
+                    limit=DISCOVERY_BATCH_SIZE, _continue=continue_ref
+                )
+                pod_items = pods.items
```

Comment:
add node_requests in this loop since pod_items gets overwritten and we only need a little data in the pod for the node

Reply:
also might want to try to match pod to job here too

Reply:
fixed